website/src/components/Tasks/EvaluateTask.tsx

```diff
@@ -18,29 +20,40 @@ export const EvaluateTask = ({
   onValidityChanged,
 }: TaskSurveyProps<RankTaskType, EvaluateTaskReply>) => {
   const cardColor = useColorModeValue("gray.50", "gray.800");
-  const [ranking, setRanking] = useState<number[]>(null);
-
-  let messages = [];
+  const [ranking, setRanking] = useState<number[] | null>(null);
+  const [notRankable, setNotRankable] = useState(false);
+  let messages: Message[] = [];
   if (task.type !== TaskType.rank_initial_prompts) {
     messages = task.conversation.messages;
   }
 
   useEffect(() => {
     if (ranking === null) {
       if (task.type === TaskType.rank_initial_prompts) {
-        onReplyChanged({ ranking: task.prompts.map((_, idx) => idx) });
+        onReplyChanged({ ranking: task.prompts.map((_, idx) => idx), not_rankable: notRankable });
       } else {
-        onReplyChanged({ ranking: task.replies.map((_, idx) => idx) });
+        onReplyChanged({ ranking: task.replies.map((_, idx) => idx), not_rankable: notRankable });
+      }
+      if (!notRankable) {
+        onValidityChanged("DEFAULT");
       }
-      onValidityChanged("DEFAULT");
     } else {
-      onReplyChanged({ ranking });
+      onReplyChanged({ ranking, not_rankable: notRankable });
       onValidityChanged("VALID");
     }
-  }, [task, ranking, onReplyChanged, onValidityChanged]);
+    if (!notRankable) {
+      onValidityChanged("VALID");
+    }
+  }, [task, ranking, onReplyChanged, onValidityChanged, notRankable]);
 
-  const sortables = task.type === TaskType.rank_initial_prompts ? "prompts" : "replies";
+  const handleNotRankableChange = useCallback((e: ChangeEvent<HTMLInputElement>) => {
+    setNotRankable(e.target.checked);
+  }, []);
 
+  const { t } = useTranslation("tasks");
+  // @notmd: I haven't test `rank_initial_prompts` type yet
```

Comment:
what do you mean by this?

Reply:
It means I haven't tested the `rank_initial_prompts` task yet. We currently don't enable that task in the backend.

Reply:
Currently there are no plans to re-enable initial prompt ranking.